src/HPWHpresets.cc

```diff
@@ -68,7 +51,7 @@ int HPWH::initResistanceTank(double tankVol_L,
     doTempDepression = false;
     tankMixesOnDraw = true;
 
-    HeatSource resistiveElementBottom(this);
+    HeatSource resistiveElementBottom(this, get_courier());
```

Comment:
I wonder if we can make a method on HPWH to "add_heat_source" that takes care of assigning the pointer and the courier. It could return a pointer to the heat source so it can be further modified instead of creating the whole thing and copying it into the HPWH heatSources vector?

Reply:
Added function HPWH::makeHeatSource.